src/ssl/support.cc

```diff
@@ -926,6 +926,11 @@ Ssl::verifySslCertificate(Security::ContextPointer &ctx, CertificateProperties c
 bool
 Ssl::setClientSNI(SSL *ssl, const char *fqdn)
 {
+    URL tmpUrl;
```

Comment:
A URL object is rather expensive to create. Ideally, the "is this an IP address?" test should be factored out, but doing so correctly is difficult, so let's just use what hostIsNumeric() (and other code) does. Something like this may work:

```
const Ip::Address test(fqdn);
if (!test.isAnyAddr())
    return; // raw IP is inappropriate for SNI
```